x-pack/plugins/lens/public/indexpattern_datasource/datapanel.tsx

```diff
@@ -347,23 +347,7 @@ export const InnerIndexPatternDataPanel = function InnerIndexPatternDataPanel({
       supportedFieldTypes.has(field.type)
     );
     const sorted = allSupportedTypesFields.sort(sortFields);
-    let groupedFields;
-    // optimization before existingFields are synced
-    if (!hasSyncedExistingFields) {
-      groupedFields = {
```

Comment:
this code is almost a complete copy of what is below. Which, moreover, is redefined below